### DISCUSSION TITLE
Add support for suffixes in formatters

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5765] [Feature Requests]
If formatter definition includes a `<field>` tag with a `sep` attribute, but without the actual field name, the separator is not included in the output.

Since separator is only included before the field value, this restricts ability to add suffixes to formatted output.

The workaround is to add a `<field>` tag with a field name that is likely to always be empty.

- [ ] Consider changing the behavior to include separator even if field name was not provided
- [ ] Whatever behavior is decided on, make sure it's consistent between front-end formating and back-end formatting

**Comment by grantfitzsimmons** [comment]
From @bronwyncombs in https://github.com/specify/specify7/issues/5077:

> **Is your feature request related to a problem? Please describe.**
> 
> You are unable to add a separator value that goes after a field when defining record formats. A common example/need for this is to add parenthesis around a field within the format. 
> 
> > Is it possible to add brackets in the table format as a separator?
> >  ![image](https://github.com/specify/specify7/assets/135047322/15189411-4b39-4885-9744-0b5a5e0ff49f)
> 
> 
> 
> **Describe the solution you'd like**
> A clear and concise description of what you want to happen.
> 
> Allow for defining a blank/null field with separator that shows up
> i.e.
> ```
> <field sep="( ">address2</field>
> <field sep=") ">blank</field>
> ```
> 
> **Describe alternatives you've considered**
> A clear and concise description of any alternative solutions or features you've considered.
> 
> Allow for defining a suffix separator that goes in-line with the field 
> i.e.
> `<field sep="( ">address2<sep=") "</field>`
> 
> **Reported By**
> Jordi @ MCNB
> 
> **Additional context**
> Add any other context or screenshots about the feature request here.
> 
> The only way you can achieve this currently is if there is another field following the field that needs to be enclosed. The following field must have a value on the record to appear, so if that field is null the parentheses would be null for the first field. 
> 
> <img width="769" alt="Screenshot 2024-07-08 at 8 45 29 AM" src="https://github.com/specify/specify7/assets/135047322/e12a08ff-efe3-4796-832c-f1cc205b6688">
> 
> 

From @jagullov:

> Thank you for taking the request. it's just to follow the recommendations as indicated by DarwinCore https://dwc.tdwg.org/list/#dwc_preparations 
![image](https://github.com/specify/specify7/assets/15903888/dcb5c3ec-1226-4f44-a3f5-69b8a4b57d48)
